cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -291,6 +310,19 @@ public void addOutput(String datasetName, Map<String, String> arguments) {
 
   @Override
   public void addOutput(String outputName, OutputFormatProvider outputFormatProvider) {
+    try {
```

Comment:
same as for input, here you create an externalDataset for every existing fileset that is used as output in MR.